### ISSUE TITLE
include `check_column_has_no_na` in `r2dii.utils`

### ISSUE THREAD

**Issue by jdhoffa** [2DegreesInvesting/r2dii.utils#21]
I find myself using the following function frequently in `r2dii.analysis` to ensure that no crucial input columns contain any NA's: 

```
check_column_has_no_na <- function(data, column) {
  if (anyNA(data[[column]])) {
    rlang::abort(
      class = "column_has_na",
      glue::glue("Column `{column}` must not contain any `NA`s.")
    )
  }

  invisible(data)
}
```

**Comment by maurolepore** [comment]
Before working on this I would like to learn more about existing frameworks and discuss when to use them versus when to write custom assertions. It seems a great topic for a ds-incubator meetup (https://github.com/2DegreesInvesting/ds-incubator/issues/11#issuecomment-606093469), and maybe @vintented and @looreen might help lead it?

**Comment by cjyetman** [comment]
I'm interested in this discussion, if it's still on the table. I've done a ton of this lately (custom assertions and error reporting).

**Comment by maurolepore** [comment]
Absolutely. You've been thinking a lot about 2DII's data and it'd be great to advertise your work and similar (e.g. https://cran.r-project.org/web/packages/assertr/vignettes/assertr.html).

What would be your preferred way to revive this conversation and engage our colleagues?

**Comment by cjyetman** [comment]
At the moment, I'd be more interested in a discussion with some at least minimally informed people about this topic. I think by now I'm particularly aware of the main options available, but in my use case none of them met all of my needs. I'd be interested in hearing about other available options and/or other needs that other people have had that were not met by current options. I'm also happy to discuss my self-made implementation of this, what it does, why I needed it, and possibly how it could be generalized to meet other needs.... but also would be happy to find an alternative that someone else maintains. 

In the long run I'd be happy to do/help with a presentation of learnings/decisions about this topic, but at the moment I feel like I'm more at the discussion phase. I did do a DS-incubator early last summer about using `assertr`, but I probably know a lot more about this by now. 😉

**Comment by maurolepore** [comment]
Good. How about starting with a call with whoever you think might help at this stage? I thought of starting a conversation in the "Discussions" tab of some repo -- but what repo should that be?

**Comment by cjyetman** [comment]
Let's have a call and/or start a Slack channel early in the new year to start.
